### PR TITLE
[OBSDEF-7152] Add symptomID and remedy for preupdate checks

### PR DESCRIPTION
**Pull request by tmarcu**
Signed-off-by: Tudor Marcu <Tudor.Marcu@emc.com>

## Purpose
[OBSDEF-7152](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-7152)

## PR checklist
- [ ] make test passed
- [ ] make build passed
- [ ] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed
- [ ] Passed - https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/

## Testing
_Paste URL of charts-custom-ci job here_

_Paste the output of your helm install/vSphere7 deployment testing here_

### CODE REVIEW

File: decks/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+- description: "Pre-Update Health Check for Application"
+  name: "Pre-Update"
+  issueCategory: Auto
+  matchOnList:
+    - matchon:
+        - label: SymptomID
```
Comment (vgupta-mickey) [inline decks/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml:6]: Each application should have unique SymptomID

File: ecs-cluster/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+- description: "Pre-Update Health Check for Application"
+  name: "Pre-Update"
+  issueCategory: Auto
+  matchOnList:
+    - matchon:
+        - label: SymptomID
+          value: OBJSC-HC-1000
```
Comment (vgupta-mickey) [inline ecs-cluster/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml:7]: should be different

File: kahm/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+- description: "Pre-Update Health Check for Application"
+  name: "Pre-Update"
+  issueCategory: Auto
+  matchOnList:
+    - matchon:
+        - label: SymptomID
+          value: OBJSC-HC-1000
```
Comment (vgupta-mickey) [inline kahm/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml:7]: need unique symptomID

File: objectscale-manager/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+- description: "Pre-Update Health Check for Application"
+  name: "Pre-Update"
+  issueCategory: Auto
+  matchOnList:
+    - matchon:
+        - label: SymptomID
+          value: OBJSC-HC-1000
```
Comment (vgupta-mickey) [inline objectscale-manager/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml:7]: unique SymptomID

**Review by vgupta-mickey — CHANGES_REQUESTED** [review]

File: decks/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml (hunk 1 above)
Comment (tmarcu) [reply decks/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml:6]: ah I misunderstood. So maybe there should be some app identifier or should I only increment the number for other apps?

Comment (vgupta-mickey) [reply decks/alerts/issues/OBJSC-HC-1000_PRE_UPDATE_issue.yaml:6]: App identifier

File: decks/templates/decks-app-configmap.yaml [hunk 5]
```diff
@@ -52,6 +52,8 @@ data:
           container: {{.Values.global.registry}}/{{.Values.healthChecks.preUpdate.image.repository}}:{{default .Values.tag .Values.healthChecks.preUpdate.image.tag}}
           serviceaccount: {{.Release.Name}}-healthchecks
           timelimit: "5m"
+          reason: "pre-update"
+          symptomID: "OBJSC-HC-1000"
```
Comment (vgupta-mickey) [inline decks/templates/decks-app-configmap.yaml:56]: DECKS-HC-1000

File: kahm/templates/kahm-app-configmap.yaml [hunk 6]
```diff
@@ -22,6 +22,8 @@ data:
         container: {{.Values.global.registry}}/{{.Values.healthChecks.preUpdate.image.repository}}:{{default .Values.tag .Values.healthChecks.preUpdate.image.tag}}
         serviceaccount: {{.Release.Name}}-healthchecks
         timelimit: "5m"
+        reason: "pre-update"
+        symptomID: "OBJSC-HC-1000"
```
Comment (vgupta-mickey) [inline kahm/templates/kahm-app-configmap.yaml:26]: KAHM-HC-1000

Comment (vgupta-mickey) [reply kahm/templates/kahm-app-configmap.yaml:26]: under eventRules?
https://github.com/EMCECS/charts/blob/72934ed38a5ab3331458eb5f8fbf9a348c977aca/objectscale-manager/templates/objectscale-manager-app-configmap.yaml#L19-L24

File: decks/templates/decks-app-configmap.yaml (hunk 5 above)
Comment (vgupta-mickey) [reply decks/templates/decks-app-configmap.yaml:56]: https://github.com/EMCECS/charts/blob/72934ed38a5ab3331458eb5f8fbf9a348c977aca/objectscale-manager/templates/objectscale-manager-app-configmap.yaml#L19-L24

File: objectscale-manager/templates/objectscale-manager-app-configmap.yaml [hunk 7]
```diff
@@ -56,6 +56,8 @@ data:
           container: {{.Values.global.registry}}/objectscale-manager-pre-update:{{ default .Values.tag .Values.healthChecks.preUpdate.image.tag}}
           serviceaccount: {{ .Release.Name }}-healthchecks
           timelimit: "5m"
+          reason: "pre-update"
+          symptomID: "OBJSC-HC-1000"
```
Comment (vgupta-mickey) [inline objectscale-manager/templates/objectscale-manager-app-configmap.yaml:60]: OBJSC-MGR-HC-1000

**Review by vgupta-mickey — CHANGES_REQUESTED** [review]

**Review by vgupta-mickey — APPROVED** [review]
Looks good but please test before the merge

**Review by eddiepavkovic — APPROVED** [review]
ok

**Pull request merged**